models/Exhibit.php

```diff
@@ -408,39 +424,64 @@ public function getRecordUrl($action = 'show')
     }
 
     /**
-     * Get a representative file for this Exhibit.
-     *
-     * The representative is the first attached file in the exhibit.
+     * Get the specified cover image File for this exhibit.  If none exists,
+     * return null.
      *
      * @return File|null
      */
-    public function getFile()
+    public function getCoverImage()
     {
         $db = $this->getDb();
         $fileTable = $this->getDb()->getTable('File');
-        $select =
-            $fileTable->getSelect()
-            ->joinInner(
-                array('eba' => $db->ExhibitBlockAttachment),
-                'eba.file_id = files.id',
-                array()
-            )
-            ->joinInner(
-                array('epb' => $db->ExhibitPageBlock),
-                'epb.id = eba.block_id',
-                array()
-            )
-            ->joinInner(
-                array('ep' => $db->ExhibitPage),
-                'ep.id = epb.page_id',
-                array()
-            )
-            ->where('ep.exhibit_id = ?', $this->id)
-            ->where('files.has_derivative_image = 1')
-            ->order(array('ep.order', 'ep.parent_id', 'epb.order', 'eba.order'))
-            ->limit(1);
-
-        return $fileTable->fetchObject($select);
+        $file = $fileTable->find($this->cover_image_file_id);
+
+        return ($file) ? $file : null;
+    }
+
+    /**
+     * Get a representative File for this Exhibit.
+     *
+     * The representative File is the File with the id specified in
+     * cover_image_file_id. If that field is empty, the first File in the
+     * Exhibit is returned. If the Exhibit has no associated Files, null is
+     * returned.
+     *
+     * @return File|null
+     */
+    public function getFile()
+    {
+        if (isset($this->cover_image_file_id)) {
+            $file = $this->getCoverImage();
+        } elseif ($this->exists()) {
```

Comment:
This is going to cause a notice if you try to getFile on an unsaved Exhibit. Moving `$file = null` from 457 up to the top of the method would fix that.